atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/metrics/LegacySweepMetricsTest.java

```diff
@@ -0,0 +1,159 @@
+    }
+
+    @After
+    public void tearDown() {
+        AtlasDbMetrics.setMetricRegistries(new MetricRegistry(),
+                new DefaultTaggedMetricRegistry());
+    }
+
+    @Test
+    public void allMetricsAreSet() {
+        sweepMetrics.updateSweepTime(
+                SWEEP_RESULTS.getTimeInMillis(),
+                SWEEP_RESULTS.getTimeElapsedSinceStartedSweeping());
+        sweepMetrics.updateCellExaminedDeleted(EXAMINED, DELETED);
+
+        assertRecordedExaminedDeletedTime(
+                ImmutableList.of(EXAMINED, DELETED, TIME_SWEEPING)
+        );
+
+        sweepMetrics.updateSweepTime(
+                OTHER_SWEEP_RESULTS.getTimeInMillis(),
+                OTHER_SWEEP_RESULTS.getTimeElapsedSinceStartedSweeping());
+        sweepMetrics.updateCellExaminedDeleted(OTHER_EXAMINED, OTHER_DELETED);
+
+        assertRecordedExaminedDeletedTime(
+                ImmutableList.of(EXAMINED + OTHER_EXAMINED,
+                DELETED + OTHER_DELETED,
+                TIME_SWEEPING + OTHER_TIME_SWEEPING)
+        );
+    }
+
+    @Test
+    public void timeElapsedMeterIsSet() {
+        sweepMetrics.updateSweepTime(
+                SWEEP_RESULTS.getTimeInMillis(),
+                SWEEP_RESULTS.getTimeElapsedSinceStartedSweeping());
+        assertSweepTimeElapsedCurrentValueWithinMarginOfError(START_TIME);
+
+        sweepMetrics.updateSweepTime(
+                OTHER_SWEEP_RESULTS.getTimeInMillis(),
+                OTHER_SWEEP_RESULTS.getTimeElapsedSinceStartedSweeping());
+        assertSweepTimeElapsedCurrentValueWithinMarginOfError(OTHER_START_TIME);
+    }
+
+    @Test
+    public void testSweepError() {
+        sweepMetrics.sweepError();
+        sweepMetrics.sweepError();
+
+        assertThat(getCounter(AtlasDbMetricNames.SWEEP_ERROR).getCount(), equalTo(2L));
+    }
+
+    private void assertRecordedExaminedDeletedTime(List<Long> values) {
+        for (int index = 0; index < 3; index++) {
+            Counter counter = getCounter(CURRENT_VALUE_LONG_METRIC_NAMES.get(index));
+            assertThat(counter.getCount(), equalTo(values.get(index)));
+        }
+    }
+
+    private void assertSweepTimeElapsedCurrentValueWithinMarginOfError(long timeSweepStarted) {
+        Gauge<Long> gauge = getGauge(AtlasDbMetricNames.TIME_ELAPSED_SWEEPING);
+        assertWithinErrorMarginOf(gauge.getValue(), System.currentTimeMillis() - timeSweepStarted);
+    }
+
+    private Counter getCounter(String namePrefix) {
+        return metricRegistry.counter(MetricRegistry.name(LegacySweepMetrics.METRIC_BASE_NAME, namePrefix));
+    }
+
+    private Gauge<Long> getGauge(String namePrefix) {
+        return metricRegistry.getGauges().get(MetricRegistry.name(LegacySweepMetrics.METRIC_BASE_NAME, namePrefix));
+    }
+
+    private void assertWithinErrorMarginOf(long actual, long expected) {
+        assertThat(actual, greaterThan((long) (expected * .95)));
+        assertThat(actual, lessThanOrEqualTo((long) (expected * 1.05)));
```

Comment:
Not sure if the hamcrest matchers have a 'within 5%' like the assertj ones do. If they don't have one then this is fine

Reply:
apparently closest hamcrest matcher is `closeTo`; but it doesn't look as good as assertj one. This one looked a bit easier to understand to me, so I kept the old one.